scripts/release

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+"""
+A multi-command tool to automate steps of the release process
+"""
+import os
+import re
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+from typing import Any, List, Union
+
+import click
+
+
+CONTEXT_SETTINGS = dict(help_option_names=["-h", "--help"])
+
+
+ROOT_DIR = Path(__file__).parent.parent
+INIT_PATH = ROOT_DIR / "ggshield" / "__init__.py"
+CASSETTES_DIR = ROOT_DIR / "tests" / "unit" / "cassettes"
+
+VERSION = os.environ.get("VERSION", "")
+TAG = f"v{VERSION}"
+
+# The branch this script must be run from. Can be overriden with an environment
+# variable to make working on the script itself easier.
+RELEASE_BRANCH = os.environ.get("RELEASE_BRANCH", "main")
+
+
+def check_run(
+    cmd: List[Union[str, Path]], **kwargs: Any
+) -> subprocess.CompletedProcess:
+    return subprocess.run(cmd, check=True, **kwargs)
+
+
+def log_progress(message: str) -> None:
+    click.secho(message, fg="magenta", err=True)
+
+
+def fail(message: str) -> None:
+    prefix = click.style("ERROR:", fg="red")
+    click.secho(f"{prefix} {message}", err=True)
+    sys.exit(1)
+
+
+def check_working_tree_is_on_release_branch() -> None:
+    proc = check_run(["git", "branch"], capture_output=True, text=True)
+    lines = proc.stdout.splitlines()
+    if f"* {RELEASE_BRANCH}" not in lines:
+        fail(f"This script must be run on the '{RELEASE_BRANCH}' branch")
+
+
+def check_working_tree_is_clean(allow_dirty: bool) -> None:
+    proc = check_run(["git", "status", "--porcelain"], capture_output=True, text=True)
+    lines = proc.stdout.splitlines()
+    if lines:
+        if allow_dirty:
+            log_progress("Ignoring changes because --allow-dirty is set")
+        else:
+            fail("Working tree contains changes. Use --allow-dirty to ignore.")
+
+
+@click.group(context_settings=CONTEXT_SETTINGS)
+@click.option(
+    "--allow-dirty",
```

Comment:
Is there a reason for allowing this option ? Isn't it error prone to make the release with changes in the working tree ?

Reply:
👌